stetho/src/main/java/com/facebook/stetho/screencast/android/ScreencastDispatcher.java

```diff
@@ -0,0 +1,149 @@
+
+  public void startScreencast(JsonRpcPeer peer, Page.StartScreencastRequest request) {
+    mRequest = request;
+
+    LogUtil.d("Starting screencast");
+    if (mHandlerThread == null) {
+      mHandlerThread = new HandlerThread("Screencast Thread");
+      mHandlerThread.start();
+    }
+    mPeer = peer;
+    mIsRunning = true;
+    mStream = new ByteArrayOutputStream();
+    mBackgroundHandler = new Handler(mHandlerThread.getLooper());
+    mMainHandler.postDelayed(mBitmapFetchRunnable, FRAME_DELAY);
+  }
+
+  public void stopScreencast() {
+    LogUtil.d("Stopping screencast");
+    mMainHandler.removeCallbacks(mBitmapFetchRunnable);
+    mBackgroundHandler.removeCallbacks(mEventDispatchRunnable);
+    mIsRunning = false;
+    mBitmap = null;
+    mCanvas = null;
+    mStream = null;
+  }
+
+  private class BitmapFetchRunnable implements Runnable {
+    @Override
+    public void run() {
+      getScreenBitmap();
+      mBackgroundHandler.post(mEventDispatchRunnable.withEndAction(this));
+    }
+
+    private Bitmap getScreenBitmap() {
+      if (!mIsRunning) {
+        return null;
+      }
+      Activity activity = mActivityTracker.tryGetTopActivity();
+      if (activity == null) {
+        return null;
+      }
+      // This stuff needs to happen in the UI thread
+      View rootView = activity.getWindow().getDecorView();
+      try {
+        if (mBitmap == null) {
+          int viewWidth = rootView.getWidth();
+          int viewHeight = rootView.getHeight();
+          float scale = Math.min((float) mRequest.maxWidth / (float) viewWidth,
+              (float) mRequest.maxHeight / (float) viewHeight);
+          int destWidth = (int) (viewWidth * scale);
+          int destHeight = (int) (viewHeight * scale);
+          mBitmap = Bitmap.createBitmap(destWidth, destHeight, Bitmap.Config.RGB_565);
+          mCanvas = new Canvas(mBitmap);
+          Matrix matrix = new Matrix();
+          mTempSrc.set(0, 0, viewWidth, viewHeight);
+          mTempDst.set(0, 0, destWidth, destHeight);
+          matrix.setRectToRect(mTempSrc, mTempDst, Matrix.ScaleToFit.CENTER);
+          mCanvas.setMatrix(matrix);
+        }
+        rootView.draw(mCanvas);
+        return mBitmap;
+      } catch (OutOfMemoryError e) {
+        LogUtil.w("Out of memory trying to allocate screencast Bitmap.");
+        return null;
+      }
+    }
+  }
+
+  private class EventDispatchRunnable implements Runnable {
+    private Runnable mEndAction;
+
+    private EventDispatchRunnable withEndAction(Runnable endAction) {
+      mEndAction = endAction;
+      return this;
+    }
+
+    @Override
+    public void run() {
+      if (!mIsRunning || mBitmap == null) {
+        return;
+      }
+      int width = mBitmap.getWidth();
+      int height = mBitmap.getHeight();
+      mStream.reset();
+      Base64OutputStream base64Stream = new Base64OutputStream(mStream, Base64.DEFAULT);
+      mBitmap.compress(Bitmap.CompressFormat.JPEG, mRequest.quality, base64Stream);
```

Comment:
shouldn't we honor the compression type requested in the `StartScreencastRequest`? The protocol says that both JPEG and PNG are supported